outlay/app/src/test/java/app/outlay/example/LoginViewPresenterTest.java

```diff
@@ -0,0 +1,147 @@
+package app.outlay.example;
+
+import app.outlay.core.executor.DefaultSubscriber;
+import app.outlay.domain.interactor.LinkAccountUseCase;
+import app.outlay.domain.interactor.ResetPasswordUseCase;
+import app.outlay.domain.interactor.UserSignInUseCase;
+import app.outlay.domain.interactor.UserSignUpUseCase;
+import app.outlay.domain.model.Credentials;
+import app.outlay.domain.model.User;
+import app.outlay.mvp.presenter.LoginViewPresenter;
+import app.outlay.mvp.view.LoginView;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LoginViewPresenterTest {
+
+    private LoginViewPresenter loginViewPresenter;
+    private String userEmail = "user@user.com";
+    private String userPassword = "GuestGuest";
+    private User user = new User();
+    private Throwable e = new Throwable("error msg");
+
+    @Mock
+    private UserSignInUseCase userSignInUseCaseMock;
+
+    @Mock
+    private UserSignUpUseCase userSignUpUseCaseMock;
+
+    @Mock
+    private ResetPasswordUseCase resetPasswordUseCaseMock;
+
+    @Mock
+    private LinkAccountUseCase linkAccountUseCaseMock;
+
+    @Mock
+    private LoginView loginViewMock;
+
+    @Captor
+    private ArgumentCaptor<DefaultSubscriber<User>> UserSubscriberCaptor;
+
+    @Captor
+    private ArgumentCaptor<DefaultSubscriber> SubscriberCaptor;
+
+    @Before
+    public void init(){
+        loginViewPresenter = new LoginViewPresenter(userSignInUseCaseMock, userSignUpUseCaseMock, resetPasswordUseCaseMock, linkAccountUseCaseMock);
+        loginViewPresenter.attachView(loginViewMock);
+
+        user.setEmail(userEmail);
+        user.setId("1");
+    }
+
+    @Test
+    public void public_signIn(){
```

Comment:
The different verifications should be broken up a couple into smaller, shorter tests. The error case should also be a different test.